azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java

```diff
@@ -456,8 +452,11 @@ public void onError(Exception error) {
                 }
             }
 
-            if (connection != null && connection.getRemoteState() != EndpointState.CLOSED && connection.getLocalState() != EndpointState.CLOSED)
+            if (connection != null && connection.getRemoteState() != EndpointState.CLOSED && connection.getLocalState() != EndpointState.CLOSED) {
                 connection.close();
+            }
+
+            closeTask.complete(null);
```

Comment:
>closeTask.complete(null); [](start = 12, length = 25)

this is wrong. Connection.close() schedules a close call - and then this line notifies users CompletableFuture - that Close is done. But, the truth is its in progress.

Reply:
We fixed several bugs in the past (#5) to clean up all the local resources, by the time close call returns. This change effectively undo's them.
`MessagingFactory` holds on to the `ReactorThread` - until the response from the service is returned. We free up the localresources - like nio.socket & reactor Thread once the service returns or the closeTask times out. 

Again, can you please explain - what is the value add here by removing this functionality. Why can't we let users, who do not care about this - to fire a close() call and not await on the call?